runelite-client/src/main/java/net/runelite/client/plugins/combatnotifier/CombatNotifier.java

```diff
@@ -0,0 +1,88 @@
+package net.runelite.client.plugins.combatnotifier;
+
+import net.runelite.api.Actor;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.Player;
+import net.runelite.client.RuneLite;
+import net.runelite.client.plugins.Plugin;
+
+import java.awt.*;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class CombatNotifier extends Plugin
+{
+	private static final Logger logger = LoggerFactory.getLogger(CombatNotifier.class);
+
+	private static final String OPERATING_SYSTEM = System.getProperty("os.name");
+	private static final int CHECK_INTERVAL = 1;
+
+	private final Client client = RuneLite.getClient();
+	private final TrayIcon trayIcon = RuneLite.getTrayIcon();
+	private final CombatNotifierConfig config = RuneLite.getRunelite().getConfigManager()
+			.getConfig(CombatNotifierConfig.class);
+
+	private ScheduledFuture<?> future;
+	private Instant lastInteracting;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		ScheduledExecutorService executor = RuneLite.getRunelite().getExecutor();
+		future = executor.scheduleAtFixedRate(this::checkIdle, CHECK_INTERVAL, CHECK_INTERVAL, TimeUnit.SECONDS);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		future.cancel(true);
+	}
+
+	private void checkIdle()
+	{
+		if (client.getGameState() != GameState.LOGGED_IN || !config.isEnabled())
+		{
+			return;
+		}
+
+		Actor opponent = getOpponent();
+		if (opponent != null)
+		{
+			if (opponent.getCombatLevel() > 1)
+			{
+				lastInteracting = Instant.now();
+			}
+		}
+
+		Duration waitDuration = Duration.ofMillis(config.getTimeout());
+		if (lastInteracting != null && Instant.now().compareTo(lastInteracting.plus(waitDuration)) >= 0)
+		{
+			trayIcon.displayMessage("RuneLite", "[" + client.getLocalPlayer().getName() + "] is now out of combat!", TrayIcon.MessageType.NONE);
+
+			if (OPERATING_SYSTEM.startsWith("Windows"))
```

Comment:
I feel like this logic should be a part of some core tray icon functionality since the idle animation plugin uses it too